src/App.tsx

```diff
@@ -1,27 +1,70 @@
-import React from 'react';
+import { FC, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { get5First, getAll, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+enum SortType {
+  ALL,
+  FIRST5,
+  RED,
+}
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+export const App: FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [sortType, setSortType] = useState<SortType>();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const getGoods = async (promise: Promise<Good[]>): Promise<void> => {
+    setGoods(await promise);
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => {
+          if (sortType !== SortType.ALL) {
+            setSortType(SortType.ALL);
+            getGoods(getAll());
+          }
```

Comment:
can we create some type of map? SortType <-> Methods to create universal handler?